components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1101,15 +1102,32 @@ void KeyringService::CreateWallet(const std::string& mnemonic,
   }
 
   ResetAllAccountInfosCache();
-  std::move(callback).Run(mnemonic);
+
+  WalletDataFilesInstaller::GetInstance()
+      .MaybeRegisterWalletDataFilesComponentOnDemand(base::BindOnce(
+          [](const std::string& mnemonic, CreateWalletCallback callback) {
+            std::move(callback).Run(mnemonic);
+          },
+          mnemonic, std::move(callback)));
 }
```

Comment:
How much time will it take to setup component?
Do we need some spinner with label like "Wallet is being initialized" or so?

Reply:
Yes, we have spinner, desktop commit https://github.com/brave/brave-core/pull/20741/commits/6d77b80c9d5525b81e6ff3adf336a0a5fb618d0d.
Android too in this PR, iOS UI PR will be covered by https://github.com/brave/brave-ios/issues/8345.